mlir/lib/Target/LLVMIR/Dialect/GPU/SelectObjectAttr.cpp

```diff
@@ -377,10 +394,21 @@ llvm::LaunchKernel::createKernelLaunch(mlir::gpu::LaunchFuncOp op,
   if (!binary)
     return op.emitError() << "Couldn't find the binary: " << binaryIdentifier;
 
+  llvm::Constant *paramsCount =
+      llvm::ConstantInt::get(i64Ty, op.getNumKernelOperands());
+
+  std::string binarySizeIdentifier = getBinarySizeIdentifier(moduleName);
+  Value *binarySizeVar = module.getGlobalVariable(binarySizeIdentifier, true);
```

Comment:
Why can't you use the initializer in `Value *binary = module.getGlobalVariable(binaryIdentifier, true);` to obtain the size of the binary?

Reply:
Done.